omni-relayer/src/workers/evm.rs

```diff
@@ -33,32 +41,117 @@ pub async fn finalize_withdraw(
 
         let mut handlers = Vec::new();
         for (key, event) in events {
-            if let Ok(withdraw_log) = serde_json::from_str::<Vec<u8>>(&event) {
+            if let Ok((block_number, log, tx_logs)) =
+                serde_json::from_str::<(u64, Log, Option<TransactionReceipt>)>(&event)
+            {
+                let Ok(light_client_latest_block_number) =
+                    utils::near::get_eth_light_client_last_block_number(&config, &jsonrpc_client)
+                        .await
+                else {
+                    warn!("Failed to get eth light client last block number");
+                    continue;
+                };
+
+                if block_number > light_client_latest_block_number {
+                    continue;
+                }
+
+                let Ok(init_log) = log.log_decode::<utils::evm::InitTransfer>() else {
+                    warn!("Failed to decode log as InitTransfer: {:?}", log);
+                    continue;
+                };
+
                 handlers.push(tokio::spawn({
+                    let config = config.clone();
                     let mut redis_connection = redis_connection.clone();
                     let connector = connector.clone();
+                    let jsonrpc_client = jsonrpc_client.clone();
 
                     async move {
-                        let Ok(fin_transfer_args) =
-                            borsh::from_slice::<FinTransferArgs>(&withdraw_log)
+                        info!("Received InitTransfer log");
+
+                        let Some(tx_hash) = log.transaction_hash else {
+                            warn!("No transaction hash in log: {:?}", log);
+                            return;
+                        };
+
+                        let Some(topic) = log.topic0() else {
+                            warn!("No topic0 in log: {:?}", log);
+                            return;
+                        };
+
+                        let tx_hash = H256::from_slice(tx_hash.as_slice());
+
+                        let Ok(token) = init_log.inner.token.parse::<AccountId>() else {
+                            warn!(
+                                "Failed to parse token as AccountId: {:?}",
+                                init_log.inner.token
+                            );
+                            return;
+                        };
+                        let Ok(recipient) = init_log.inner.recipient.parse::<AccountId>() else {
+                            warn!(
+                                "Failed to parse recipient as AccountId: {:?}",
+                                init_log.inner.recipient
+                            );
+                            return;
+                        };
+
+                        let vaa = utils::evm::get_vaa(tx_logs, &log, &config).await;
```

Comment:
In case of wormhole we don't need to wait for block number on EthLightClient.

Reply:
Fixed in 8b469bc